src/logic/hooks/useEstimateSafeTxGas.tsx

```diff
@@ -0,0 +1,49 @@
+import { Operation } from '@gnosis.pm/safe-react-gateway-sdk'
+import { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+
+import { estimateSafeTxGas } from 'src/logic/safe/transactions/gas'
+import { currentSafe } from 'src/logic/safe/store/selectors'
+
+type UseEstimateSafeTxGasProps = {
+  isCreation: boolean
+  txData: string
+  txRecipient: string
+  txAmount: string
+  operation?: Operation
+}
+
+export const useEstimateSafeTxGas = ({
+  isCreation,
+  txData,
+  txRecipient,
+  txAmount,
+  operation,
+}: UseEstimateSafeTxGasProps): string => {
+  const [safeTxGasEstimation, setSafeTxGasEstimation] = useState<string>('0')
+  const { address: safeAddress, currentVersion: safeVersion } = useSelector(currentSafe) ?? {}
+
+  useEffect(() => {
+    if (!isCreation) return
```

Comment:
So we're passing `isCreation` just to not run the hook, essentially?

Reply:
If its an immediate execution, `isCreation` should be true. Alternatively we could handle it more implicit and only do the estimation if `safeTxGas` doesn't exist since the only place it exists is the `ApproveTxModal` where its not a creation.

Reply:
Oh, wait, I misread it. So we only estimate safeTxGas on creation, and never again? Sound correct.